src/display.rs

```diff
@@ -359,32 +469,71 @@ impl<'a, 'b> State<'a, 'b> {
                     bounds,
                 ))
             }
-            Data::Union(_) => {
-                let meta = self.find_meta(&self.input.attrs)?.ok_or_else(|| {
+            syn::Data::Union(_) => {
+                let meta = self.find_meta(&self.input.attrs, "fmt")?.ok_or_else(|| {
                     Error::new(
                         self.input.span(),
                         "Can not automatically infer format for unions",
                     )
                 })?;
-                let fmt = self.get_meta_fmt(&meta, false)?.0;
+                let fmt = self.parse_meta_fmt(&meta, false)?.0;
 
                 Ok((
                     quote_spanned!(self.input.span()=> _ => write!(_derive_more_Display_formatter, "{}", #fmt),),
                     HashMap::new(),
                 ))
             }
-        }
+        };
+
+        let (arms, mut bounds) = result?;
+
+        match self.find_meta(&self.input.attrs, "bound")? {
```

Comment:
Can you change this to early returns too?

Reply:
Also done.